### PR TITLE
Upgrade dependencies to Ember 3.20

### PR DESCRIPTION
**Pull request by GreatWizard**
### Upgrade dependencies to Ember 3.20 (#56)

- Update dependencies and code to fit Ember 3.20
- Use qunit-dom asserts
- Update CircleCI config
- Rework linter rules

### CODE REVIEW

File: tests/integration/helpers/remove-diacritics-test.js [hunk 1]
```diff
@@ -8,9 +8,9 @@ module('Integration | Helper | remove-diacritics', function(hooks) {
 
   test('it removes diacritics', async function(assert) {
     await render(hbs`{{remove-diacritics ''}}`)
-    assert.equal(this.element.textContent.trim(), '')
+    assert.dom(this.element).hasText('')
```
Comment (MrChocolatine) [inline tests/integration/helpers/remove-diacritics-test.js:11]:
```suggestion
    assert.dom().hasText('')
```

File: tests/integration/helpers/remove-diacritics-test.js [hunk 2]
```diff
@@ -8,9 +8,9 @@ module('Integration | Helper | remove-diacritics', function(hooks) {
...
     await render(hbs`{{remove-diacritics 'é è à ù ç č Ž ñ'}}`)
-    assert.equal(this.element.textContent.trim(), 'e e a u c c Z n')
+    assert.dom(this.element).hasText('e e a u c c Z n')
```
Comment (MrChocolatine) [inline tests/integration/helpers/remove-diacritics-test.js:13]:
```suggestion
    assert.dom().hasText('e e a u c c Z n')
```

File: tests/integration/helpers/slugify-test.js [hunk 3]
```diff
@@ -8,13 +8,13 @@ module('Integration | Helper | slugify', function(hooks) {
...
+    assert.dom(this.element).hasText('')
     await render(hbs`{{slugify 'bonjour'}}`)
-    assert.equal(this.element.textContent.trim(), 'bonjour')
+    assert.dom(this.element).hasText('bonjour')
     await render(hbs`{{slugify 'annyǒng hashimnikka'}}`)
-    assert.equal(this.element.textContent.trim(), 'annyong-hashimnikka')
+    assert.dom(this.element).hasText('annyong-hashimnikka')
     await render(hbs`{{slugify 'Bon jOuR Töi 1337   '}}`)
-    assert.equal(this.element.textContent.trim(), 'bon-jour-toi-1337')
+    assert.dom(this.element).hasText('bon-jour-toi-1337')
```
Comment (MrChocolatine) [inline tests/integration/helpers/slugify-test.js:17]: `assert.dom().hasText(...)`

File: config/environment.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 'use strict'
 
 module.exports = function(/* environment, appConfig */) {
-  return {}
+  return { }
```
[the remark this answers is not in the document]
Comment (GreatWizard) [reply config/environment.js:4]: No it's not needed at all. That's how the ember-cli blueprint for addon is generated

**Review by MrChocolatine — APPROVED** [review]

**Pull request merged**